tests/testthat/test-reporter.R

```diff
@@ -77,4 +82,71 @@ test_that("reporters produce consistent output", {
   save_report("silent")
   save_report("rstudio")
   save_report("junit", reporter = createJunitReporterMock())
+
+  # Test that MultiReporter can write to two different places
+  tap_file <- tempfile()
+  save_report("summary", reporter = MultiReporter$new(list(
+    SummaryReporter$new(show_praise = FALSE, omit_dots = FALSE),
+    TapReporter$new(file=tap_file))
+  ))
+  expect_identical(read_lines(tap_file),
+    read_lines(test_path("reporters", "tap.txt")))
+})
+
+
+test_report_to_file <- function(name, reporter = find_reporter_one(name, ...),
+                                output_file, ...) {
+  # output_file is where we expect output to be written to, whether we pass it
+  # as an argument to Reporter$new() (here, via the ...), or whether it is set
+  # in an option.
+  path <- test_path("reporters", paste0(name, ".txt"))
+  expect_silent(test_reporter(reporter))
+  expect_identical(read_lines(output_file), read_lines(path))
+}
+
+output <- tempfile()
```

Comment:
Could you mimic the approach of the "reporters produce consistent output" test? I think it's better to have one test, with some copy and paste inside of it. It's too hard to eliminate all the repetition here because of the varying arguments to the reporter.